pseudocontainers_task/student/src/impl.cpp

```diff
@@ -1 +1,22 @@
 #include "impl.h"
+
+#include <bitset>
+#include <iostream>
+
+std::tuple<worker_id, worker_id, days> getMaxIntersection(const std::unordered_map<worker_id, calendar>& workerVacations)
+{
+    std::tuple<worker_id, worker_id, days> intersection(0, 0, 0);
+    for (auto iW = workerVacations.cbegin(); iW != workerVacations.cend(); ++iW) {
```

Comment:
use range for

Reply:
The main idea of using iterators in these for loops is to avoid overlapping of comparison and reduce O(n*n).
external for (i: 0..end() - 1) 
internal for (j: i + 1, end())
But using iterators leads to O(n*(n+1)/2) ~~ O(n*n)
So, "for range cycle" has almost the same complexity.

Reply:
Please don't make premature optimizations that affects readability

Reply:
Upd: but, for internal loop It's okay here to do in such way. 
Sorry, my bad, internal loop can use iterators.